src/rgw/rgw_acl.h

```diff
@@ -214,20 +217,25 @@ struct ACLReferer {
       perm(perm) {
   }
 
-  bool is_match(std::string http_referer) const {
-    if (http_referer == url_spec) {
+  bool is_match(boost::string_ref http_referer) const {
+    const auto http_host = get_http_host(http_referer);
+    if (!http_host) {
+      return false;
+    }
+
+    boost::string_ref host_spec(url_spec);
```

Comment:
Hmm, I'm not sure whether we really need the `host_spec` temporary. IIUC `boost::string_ref` has the `operator==` overload for `std::string` so we could compare  `*http_host` directly with `url_spec`.